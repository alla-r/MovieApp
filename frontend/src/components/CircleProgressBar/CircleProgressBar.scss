.CircularProgressbar .CircularProgressbar-text {
  font-weight: 600;
  font-size: 24px;
  line-height: 15px;
  dominant-baseline: middle;
  text-anchor: middle;
}