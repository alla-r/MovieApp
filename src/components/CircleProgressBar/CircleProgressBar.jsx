import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import { getBarColor } from '../../global/helpers';
import theme from '../../theme';
import './CircleProgressBar.scss';

function CircleProgressBar({ voteAvg, percentage }) {
  return (
    <CircularProgressbar
      value={percentage}
      text={Number.isInteger(voteAvg) ? `${voteAvg}.0` : voteAvg.toFixed(1)}
      strokeWidth={4}
      background
      backgroundPadding={4}
      styles={buildStyles({
        backgroundColor: `${theme.colors.dark}`,
        textColor: `${theme.colors.light}`,
        pathColor: `${getBarColor('path', percentage)}`,
        trailColor: `${getBarColor('trail', percentage)}`,
      })}
    />
  );
}

CircleProgressBar.propTypes = {
  voteAvg: PropTypes.number.isRequired,
  percentage: PropTypes.number.isRequired,
};

export default CircleProgressBar;
