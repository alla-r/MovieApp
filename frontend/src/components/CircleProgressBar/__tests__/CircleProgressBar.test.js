import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import CircleProgressBar from '../CircleProgressBar';

describe('<CircleProgressBar />', () => {
  it('should be defined', () => {
    const wrapper = mount(<CircleProgressBar voteAvg={5} percentage={52} />)

    expect(wrapper).toBeDefined();
  });

  it('should match snapshot with float number', () => {
    const wrapper = mount(<CircleProgressBar voteAvg={5.2} percentage={52} />)
    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

