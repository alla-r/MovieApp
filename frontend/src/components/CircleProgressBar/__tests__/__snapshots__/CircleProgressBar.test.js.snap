// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleProgressBar /> should match snapshot with float number 1`] = `
<svg
  className="CircularProgressbar "
  data-test-id="CircularProgressbar"
  style={Object {}}
  viewBox="0 0 100 100"
>
  <circle
    className="CircularProgressbar-background"
    cx={50}
    cy={50}
    r={50}
    style={
      Object {
        "fill": "#1A191C",
      }
    }
  />
  <path
    className="CircularProgressbar-trail"
    d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
    fillOpacity={0}
    strokeWidth={4}
    style={
      Object {
        "stroke": "rgba(255, 203, 71, 0.3)",
        "strokeDasharray": "276.46015351590177px 276.46015351590177px",
        "strokeDashoffset": "0px",
      }
    }
  />
  <path
    className="CircularProgressbar-path"
    d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
    fillOpacity={0}
    strokeWidth={4}
    style={
      Object {
        "stroke": "rgba(255, 203, 71, 1)",
        "strokeDasharray": "276.46015351590177px 276.46015351590177px",
        "strokeDashoffset": "132.70087368763285px",
      }
    }
  />
  <text
    className="CircularProgressbar-text"
    style={
      Object {
        "fill": "#EAEAEA",
      }
    }
    x={50}
    y={50}
  >
    5.2
  </text>
</svg>
`;
